File: documentation-examples-java/src/java/dev/typr/foundations/docs/sqlserver/DomainType.java
package dev.typr.foundations.docs.sqlserver;

import dev.typr.foundations.SqlServerType;
import dev.typr.foundations.SqlServerTypes;

@SuppressWarnings("unused")
public class DomainType {
    //start
    // Wrapper type
    public record OrderId(Integer value) {}

    // Create SqlServerType from INT
    SqlServerType<OrderId> orderIdType =
        SqlServerTypes.int_.bimap(OrderId::new, OrderId::value);
    //stop
}

File: documentation-examples-kotlin/src/kotlin/dev/typr/foundations/docs/oracle/DomainType.kt
package dev.typr.foundations.docs.oracle

import dev.typr.foundationskt.*
import dev.typr.foundationskt.data.*

@Suppress("unused")
class DomainType {
    //start
    // Wrapper type
    data class EmployeeId(val value: Long)

    // Create OracleType from NUMBER
    val empIdType: OracleType<EmployeeId> =
        OracleTypes.numberLong.transform(::EmployeeId, EmployeeId::value)
    //stop
}
